multiqc/modules/kraken/kraken.py

```diff
@@ -295,9 +384,70 @@ def top_five_barplot(self):
                 Note that this is only an approximation, and that kraken percentages don't always add to exactly 100%.
 
                 The category _"Other"_ shows the difference between the above total read count and the sum of the read counts
-                in the top 5 taxa shown + unclassified. This should cover all taxa _not_ in the top 5, +/- any rounding errors.
+                in the top {self.top_n} taxa shown + unclassified. This should cover all taxa _not_ in the top {self.top_n}, +/- any rounding errors.
 
                 Note that any taxon that does not exactly fit a taxon rank (eg. `-` or `G2`) is ignored.
             """,
             plot=bargraph.plot(pd, cats, pconfig),
         )
+
+    def top_five_duplication_heatmap(self):
+        """ Add a heatmap showing the minimizer duplication top-5 species"""
+
+        pd = []
+        duplication = list()
+        pconfig = {"id": "kraken-topfive-duplication_plot", "title": f"Kraken 2: Top {self.top_n} species duplication"}
+
+        rank_code = "S"
+        rank_data = dict()
+        # Loop through the summed tax percentages to get the top 5 across all samples
+        try:
+            sorted_pct = sorted(self.kraken_total_pct[rank_code].items(), key=lambda x: x[1], reverse=True)
+        except KeyError:
+            pass
+            # Taxa rank not found in this sample
+
+        i = 0
+        counts_shown = {}
+
+        showed_warning = False
+        for classif, pct_sum in sorted_pct:
+            i += 1
+            if i > self.top_n:
+                break
+            # Pull out counts for this rank + classif from each sample
+            for s_name, d in self.kraken_raw_data.items():
+                sample_duplication = list()
+                if s_name not in rank_data:
+                    rank_data[s_name] = dict()
+                if s_name not in counts_shown:
+                    counts_shown[s_name] = 0
+                for row in d:
+                    if row["rank_code"] == rank_code:
+                        if row["classif"] == classif:
+                            if classif not in rank_data[s_name]:
+                                rank_data[s_name][classif] = 0
+                            try:
+                                rank_data[s_name][classif] = row["minimizer_duplication"]
+                            except KeyError:
+                                del rank_data[s_name]
```

Comment:
Updated duplication plot to remove samples from kraken report without distinct minimizer information